draco/core/io.py

```diff
@@ -713,42 +690,69 @@ class Truncate(task.SingleTask):
         Relative precision to truncate to (default 1e-4).
     variance_increase : float
         Maximum fractional increase in variance from numerical truncation.
+    ensure_chunked : bool
+        If True, ensure datasets are chunked according to their dataset_spec.
     """
 
     dataset = config.Property(proptype=list, default=None)
     weight_dataset = config.Property(proptype=list, default=None)
     fixed_precision = config.Property(proptype=float, default=None)
     variance_increase = config.Property(proptype=float, default=None)
+    ensure_chunked = config.Property(proptype=bool, default=True)
+
+    default_params = {
+        "weight_dataset": None,
+        "fixed_precision": 1e-4,
+        "variance_increase": 1e-3,
+    }
+
+    @functools.lru_cache()
+    def _get_params(self, container, dset):
+        """
+        Load truncation parameters for a dataset from config or container defaults.
+
+        Parameters
+        ----------
+        container
+            Container class.
+        dset : str
+            Dataset name
+
+        Returns
+        -------
+        Dict or None
+            Returns `None` if the dataset shouldn't get truncated.
+        """
+        # Check if dataset should get truncated at all
+        if self.dataset is None:
+            if dset not in container._dataset_spec or not container._dataset_spec[
```

Comment:
I wonder if you want to pull the dataset spec from the class rather than the instance, i.e. `container.__class__._dataset_spec`

Reply:
Why?

Reply:
I'm not quite sure what I was getting at in here, but my best interpretation 6 months later is that it's debatable if this should get the class or the instance passed in to this method.